sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/datastore/DatastoreV1.java

```diff
@@ -1409,6 +1411,7 @@ public int nextBatchSize(long timeSinceEpochMillis) {
         Metrics.counter(DatastoreWriterFn.class, "datastoreRpcErrors");
     private final Counter rpcSuccesses =
         Metrics.counter(DatastoreWriterFn.class, "datastoreRpcSuccesses");
+    private final Counter duplicateKeys = Metrics.counter(DatastoreWriterFn.class, "duplicateKeys");
```

Comment:
I worry about what users may think when they see this in their UI. They may think that we're duplicating data, although that's not exactly what's happening.

Could we instead call this metric 'earlyFlushes' or something like that?

Another interesting idea would be to have a `Distribution` metric that aggregates batch sizes instead of the earlyFlushes. This may be interesting for users debugging their transforms; and I think it should be enough to run the verifications for the test? WDYT? I think this would be ideal